File: Java/src/771_JewelsAndStones/JewelsAndStones.java
import java.util.HashMap;

class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        HashMap<Character, Integer> stone = new HashMap<>();
        char[] str = stones.toCharArray(); 
        for (char c : str) { 
            if (stone.containsKey(c)) { 
                stone.put(c, stone.get(c) + 1); 
            } 
            else { 
                stone.put(c, 1); 
            } 
         }
        int ans = 0;
        char[] strJewels = jewels.toCharArray();
        for(char c : strJewels){
            if (stone.containsKey(c))
             ans+=stone.get(c);
        }
        return ans;
    }
}

File: golang/771_JewelsAndStones/jewelsAndStones.go
package jewelsandstones

func numJewelsInStones(jewels string, stones string) int {
	ans := 0
	for i := 0; i < len(jewels); i++ {
		for j := 0; j < len(stones); j++ {
			if jewels[i] == stones[j] {
				ans++
			}
		}
	}
	return ans
}
